pkg/server/debug.go

```diff
@@ -150,6 +154,35 @@ func init() {
 `)
 	})
 
+	debugServeMux.HandleFunc(debugLogSpyEndpoint, func(w http.ResponseWriter, r *http.Request) {
+		// The explicit header is necessary or (at least Chrome) will try to
+		// download a gzipped file (Content-type comes back application/x-gzip).
+		w.Header().Add("Content-type", "text/plain") // FIXME(tschottdorf): check this is correct
+
+		ch := make(chan struct{})
+		go func() {
+			time.Sleep(5 * time.Second)
+			close(ch)
+		}()
+
+		entries := make(chan log.Entry, 100)
+
+		log.Intercept(func(finished bool, entry log.Entry) (wantMore bool) {
+			select {
+			case <-ch:
+				close(entries)
+				return false
+			default:
+				entries <- entry
+				return true
+			}
+		})
+
+		for entry := range entries {
+			fmt.Fprintln(w, entry.Message)
+		}
```

Comment:
There is probably a means of flushing each message.

Rather than explicitly filtering on range ID or other attributes, I'd add a regex which can filter what is returned. Then you can `grep -w r<someRangID>`.